sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java

```diff
@@ -59,13 +60,23 @@ public TokenResult getCachedToken(String partitionName) {
         if (token != null) {
             Calendar utcCalendar = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
 
-            /* The token is considered expired. */
+            /* Token result was cached but status was not 'succeed'. */
+            if (!token.status().equals(Constants.TOKEN_RESULT_SUCCEED)) {

```

Comment:
I do not think we should cache the failed token... I would rather not add here, if we cache the failed token.

Reply:
@yochaude You had said you were getting strange errors because we were trying to serialize failed cached tokens for iOS.  Do you know how to repro this issue on a device?